pkg/vmprovider/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -881,6 +881,36 @@ func vmTests() {
 				})
 			})
 
+			Context("VM Class Config specifies a hardware version", func() {
```

Comment:
Thanks for this test. 
- We'll need to also do the same unit test in vsphere2/vmprovider_vm_test.go. 
- Also, another thought, we could also cover the PVC/PCI device case by adding onto this test. Just by mentioning a vm.Spec.MinHardwareVersion in a BeforeEach() here (https://github.com/vmware-tanzu/vm-operator/blob/71c707d5edb2b50ccea9e6984bb599b6586e0b9a/pkg/vmprovider/providers/vsphere/vmprovider_vm_test.go#L848)